tests/Fixer/ControlStructure/YodaStyleFixerTest.php

```diff
@@ -0,0 +1,493 @@
+            ['<?php $a = 1 === 1;'],
+            ['<?php $b = $b === $c;'],
+            ['<?php $c = $$b === $$c;'],
+            ['<?php $d = count($this->array[$var]) === $a;'],
+            ['<?php $e = $a === count($this->array[$var]);'],
+            ['<?php $f = ($a & self::MY_BITMASK) === $a;'],
+            ['<?php $g = $a === ($a & self::MY_BITMASK);'],
+            ['<?php $h = $this->getStuff() === $myVariable;'],
+            ['<?php $i = $myVariable === $this->getStuff();'],
+            ['<?php $j = 2 * $myVar % 3 === $a;'],
+            ['<?php return $k === 2 * $myVar % 3;'],
+            ['<?php $l = $c > 2;'],
+            ['<?php return $this->myObject->{$index}+$b === "";'],
+            ['<?php return $m[2]+1 == 2;'],
+            // https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/693
+            ['<?php return array(2) == $o;'],
+            ['<?php return $p == array(2);'],
+            ['<?php return array($q) == $a;'],
+            ['<?php return $r == array($a);'],
+            ['<?php $s = ((array(2))) == $a;'],
+            ['<?php $t = $a == ((array(2)));'],
+            ['<?php list($a) = $c === array(1) ? $b : $d;'],
+            ['<?php $b = 7 === list($a) = [7];'],
+            ['<?php $a = function(){} === array(0);'],
+            ['<?php $z = $n == list($a) = $b;'],
+            ['<?php return $n == list($a) = $b;'],
+            // Fix cases.
+            'Array destruct by ternary.' => [
+                '<?php list($a) = 11 === $c ? $b : $d;',
+                '<?php list($a) = $c === 11 ? $b : $d;',
+            ],
+            'Less spacing.' => [
+                '<?php $z=2==$a;$b=$c>1&&$c<=10;',
+                '<?php $z=$a==2;$b=$c>1&&$c<=10;',
+            ],
+            'Comments.' => [
+                '<?php $z = /**/ /**/2/**/ /**/
+                 # aa
+                 /**/==/**/$a/***/;',
+                '<?php $z = /**/ /**/$a/**/ /**/
+                 # aa
+                 /**/==/**/2/***/;',
+            ],
+            [
+                '<?php return 2 == ($a)?>',
+                '<?php return ($a) == 2?>',
+            ],
+            [
+                '<?php $a = ($c === ((null === $b)));',
+                '<?php $a = ($c === (($b === null)));',
+            ],
+            [
+                '<?php return null == $a[2];',
+                '<?php return $a[2] == null;',
+            ],
+            [
+                '<?php return "" === $this->myArray[$index];',
+                '<?php return $this->myArray[$index] === "";',
+            ],
+            [
+                '<?php return "" === $this->myArray[$index]->a;',
+                '<?php return $this->myArray[$index]->a === "";',
+            ],
+            [
+                '<?php return "" === $this->myObject->  {$index};',
+                '<?php return $this->myObject->  {$index} === "";',
+            ],
+            [
+                '<?php return "" === $this->myObject->{$index}->a;',
```

Comment:
Please add a similar test case without the braces around the dynamic property, e.g.:
```php
<?php return $this->myObject->$index->a === "";
```